documentation/new-version-guideline.md

```diff
@@ -0,0 +1,302 @@
+		return nil, errors.New("no more items")
+	}
+	if iter.cur == nil || iter.index == len(iter.cur) {
+		// first page or page exhausted
+		page, err := iter.pager.NextPage(ctx)
+		if err != nil {
+			return nil, err
+		}
+		iter.cur = page.Items()
+		iter.index = 0
+	}
+	item := iter.cur[iter.index]
+	// advance item
+	iter.index++
+	return item, nil
+}
+
+func NewIterator[TItem any, TPage PageConstraint[TItem]](pager *runtime.Pager[TPage]) *Iterator[TItem, TPage] {
+	return &Iterator[TItem, TPage]{
+		pager: pager,
+	}
+}
+```
+
+***Usage***
+```go
+iter := NewIterator[*armresources.ResourceGroup](rgClient.NewListPager(nil))
+for iter.More() {
+    rg, err := iter.NextItem(ctx)
+    if err != nil {
+        // handle error...
+    }
+    // dealing with `rg`
+}
+```
+
+### Reference
+
+For more information, you could refer to [design guidelines of Paging](https://azure.github.io/azure-sdk/golang_introduction.html#methods-returning-collections-paging) and [API reference of pager](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime#Pager).
+
+## Long-Running Operations
+
+### General usage
+
+Some operations can take a long time to complete. Azure introduces the long-running operations (LROs) to do such operations asynchronously. You need to use Begin* to start an LRO. It will return a poller that can used to keep polling for the result until LRO is done. For examples:
+
+```go
+ctx := context.TODO()
+poller, err := client.BeginCreate(ctx, "resource_identifier", "additonal_parameter", nil)
+if err != nil {
+    // handle error...
+}
+resp, err = poller.PollUntilDone(ctx, 5 * time.Second)
+if err != nil {
+    // handle error...
+}
+// dealing with `resp`
+```
+
+> NOTE: You will need to pass a polling interval to ```PollUntilDone``` and tell the poller how often it should try to get the status. This number is usually small but it's best to consult the [Azure service documentation](https://docs.microsoft.com/azure/?product=featured) on best practices and recommended intervals for your specific use cases.
+
+### Resume Tokens
+
+Pollers provide the ability to serialize their state into a "resume token" which can be used by another process to recreate the poller. For example:
+
+```go
+ctx := context.TODO()
+poller, err := client.BeginCreate(ctx, "resource_identifier", "additonal_parameter", nil)
+if err != nil {
+    // handle error...
+}
+token, err := poller.ResumeToken()
+if err != nil {
+    // handle error...
+}
+
+// ... 
+
+// recreate the poller from the token
+poller, err = client.BeginCreate(ctx, "", "", &armresources.ResourceGroupsClientBeginCreateOptions{
+    ResumeToken: token,
+})
+resp, err = poller.PollUntilDone(ctx, 5 * time.Second)
+if err != nil {
+    // handle error...
+}
+// dealing with `resp`
+```
+
+> NOTE: A token can only be obtained for a poller that's in a non-terminal state. Each time you call poller.Poll(), you should get a new token to handle poller state change.
```

Comment:
Do we need to explain "non-terminal" state?
And I am not quite clear about the last statement here - what does it mean that we should get a new token to handle poller state change?
```suggestion
> NOTE: A token can only be obtained for a poller that's in a non-terminal state. Each time you call `poller.Poll()`, you should get a new token to handle poller state change.
```

Reply:
Each time you call `poller.Poll()`, the token might change because of the LRO state change. So if you cache the token for failover, you need to update the cache when call `poller.Poll()`.